pkg/utils/meta_loader_test.go

```diff
@@ -0,0 +1,85 @@
+package utils
+
+import (
+	"path"
+	"testing"
+
+	"github.com/mt-sre/addon-metadata-operator/internal/testutils"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	ReferenceAddonImageSetDir   = path.Join(testutils.AddonsImagesetDir, "reference-addon")
+	ReferenceAddonIndexImageDir = path.Join(testutils.AddonsIndexImageDir, "reference-addon")
+)
+
+// Version is ignored in the case of the static indexImage
+func TestMetaLoaderIndexImage(t *testing.T) {
+	expectedIndexImage := "quay.io/osd-addons/reference-addon-index@sha256:b9e87a598e7fd6afb4bfedb31e4098435c2105cc8ebe33231c341e515ba9054d"
+	cases := []struct {
+		addonDir string
+		env      string
+		version  string
+	}{
+		{
+			addonDir: ReferenceAddonIndexImageDir,
+			env:      "stage",
+			version:  "latest",
+		},
+		{
+			addonDir: ReferenceAddonIndexImageDir,
+			env:      "stage",
+			version:  "v0.0.1",
+		},
+	}
+	for _, tc := range cases {
+		tc := tc // pin
```

Comment:
I don't think this is required, all the lines following fn/methods get their own private copy.

Reply:
I forgot to add `t.Parallel()` , thanks for spotting this!

because we use `t.Run(...) + t.Parallel(...)` it spins a new goroutine in the background, sending the current context. If we don't pin `tc := tc`, the underlying reference will change in unexpected way in the goroutine that were spun off.

Reply:
Yeah now it make sense